### ISSUE TITLE
Speed & Baritone (Ouch on Blocks v.3)

### ISSUE THREAD

**Issue by DeeFeeCee** [cabaletta/baritone#382]
### What happens?

Player takes damage when "falling" off of blocks (as opposed to jumping).
For whatever reason, SPEED (Impact) causes multiple issues involving the edge between 2 blocks (including a solid block & an air block).
Here's the rundown:
1) Baritone truly doesn't know what to do with changes in speed.
----Obviously, using SPEED hacks messes up by resulting in fall damage or improper preparation, but to be clear, this happens with human controls, too (not just Baritone, but wait, there's more!).
----The pathfinding algorithm struggles to compensate for walking//running on ice, etc. & perhaps with speed potions, etc.
That is to say that Baritone doesn't plan ahead when walking on surfaces that may alter trajectory. It often tries to "backtrack", making any progress forward through speeding up pointless.
----Gets stuck between two blocks with SPEED on. To be clear, I mean on flat surfaces as well as over ledges as short as 1 block tall. (Again, an Impact thing)
----Due to the inability to compensate for higher speeds, Baritone often
--------leaves loot on the ground (mining, possibly hunting?)
--------fails to pick up water with water bucket after a planned fall (have not tested on ice, but I presume this can still happen)
----sttttttttutttters when in tight spaces (under tree leaves, in mines, etc.)
2) Using SPEED while walking off 2-high drops or higher results in fall damage (even with NO FALL on)
3) Baritone also has a strange way of pathfinding around obstructions. Although it is presumably faster to walk around a 2–high pole of cobblestone, Baritone chooses to mine both (or just the top) to continue. I haven't tried without a pick or with different blocks, but this seems silly. (This issue involves failure to calculate most efficient speed).
4) I've also noticed that when outrunning chunk rendering, Baritone just stops. This could happen with speed hacks, ice, or I guess boats & horses, too, but I heard that's a bit of a black eye for #B.
5) Sometimes when pathfinding, #B will find one focus, approach it, then find a farther one & go there instead. This is obviously not time–efficient.
6) For whatever reason, paths 🅱️ chose to make earlier (e.g. breaking a block to fall into water) are sometimes ignored afterwards (mining the block next to said hole & landing in same body of water).
7) Semi-related to speed, or perhaps a set rule, Baritone will build up to mine or approach blocks (e.g. logs in a tree) that it could reach without placing any blocks. I wonder if placing blocks should be set to a shorter time period, but with distance detection increased? Tangentially, related to 3, it'd be nice if #B took into account the pickaxe material (& possibly enchantments) or bare hand for choosing which blocks to break, if any at all. Time-to-break is given I think on the MC wiki for each block, but you can assume
----1: that if the highest speed to break a block is x, then other material pickaxes will be a function of that (.5x, .66x, .75x)
----2 (if 1 is too complex) that the average//longest time–to–break for the average block is x, then other pickaxes will be a fraction of that (~)
----3 certain blocks are just stupidly fast to break (TNT, dirt, grass), some are relatively fast (endstone, netherrack, stone), some are relatively slower (ores) &… obsidian
--------thus, if pick material is too time-consuming to account for, then a simple ratio will do (obsidian = 1, dirt =.05), or something of the sort

### Exact steps to reproduce?

Video that shows the bugs & some troubleshooting (not exhaustive, but includes as much as I deemed necessary at the time, plus video editing):
[ https://1drv.ms/v/s!AmGCvkCpn06bjbkkNWCopQj0ZdHNNQ ] (sped up for viewing pleasure)

### 0ω0 (that's zero–omega–zero, that's allowed, right?)
- Impact Version:  1.12.2 4.6
- Baritone Version: Um, I don't know. I guess Impact 4.6 has only one version…
- Using Optifine?  Yes
- Using LiteLoader?  No
- Enabled mods:  Speed, vanilla
- Operating System:  W10 64

So, these issues are all interrelated & best understood in the greater context. Apologies if this is hard to follow. I can follow up with specific points & refer to this post if needed.

**Comment by 0-x-2-2** [comment]
baritone does not work with speedhack except for timer because it just speeds up the game and this is more a impact issue with baritone not a baritone issue.

**Comment by DeeFeeCee** [comment]
I see. & with ice, etc. is there a way for Baritone to plan for acceleration?
